python/django_bolt/auth/backends.py

```
@@ -136,16 +167,40 @@ def __init__(
+        jwks_url: str | None = None,
+        jwks: dict[str, Any] | str | None = None,
     ):
+        provided_keys = [k for k in (secret, public_key) if k is not None]
+        if len(provided_keys) > 1:
+            raise ImproperlyConfigured(
+                "JWTAuthentication accepts either 'secret' (HMAC) or 'public_key' (asymmetric), not both."
+            )
+        if (jwks_url or jwks) and provided_keys:
+            raise ImproperlyConfigured(
+                "JWTAuthentication accepts JWKS (jwks_url/jwks) or a static key (secret/public_key), not both."
+            )
```

Ruff (0.15.21)

[warning] Avoid specifying long messages outside the exception class

(TRY003)